### PR TITLE
[NO TICKET] Update PyYAML and fix CI

### PR DESCRIPTION
**Pull request by lidijac**
## Proposed change
CI was broken. History of fails [here](https://github.com/getyourguide/auto-pr/actions/workflows/main.yml). I still need to see if it will publish the new version.

## Checklist

-   [ ] Tests have been added to verify that the new code works (if possible)
-   [ ] Documentation has been updated to reflect changes
-   [ ] `CHANGELOG.md` has been updated to reflect changes

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -4,11 +4,11 @@ on: [push]
 
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
```
Comment (lidijac) [inline .github/workflows/main.yml:7]: Was deprecated, didn't want to run

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -4,11 +4,11 @@ on: [push]
...
     strategy:
       matrix:
         python-version: [3.9, 3.8]
-        poetry-version: [1.1.7]
+        poetry-version: [1.3.1]
```
Comment (lidijac) [inline .github/workflows/main.yml:11]: lock file was incompatible with old version, someone prob pushed a newer one, bumped this

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ repos:
       - id: black
         additional_dependencies: ['click==8.0.4']
   - repo: https://github.com/PyCQA/isort
-    rev: 5.9.3
+    rev: 5.12.0
```
Comment (lidijac) [inline .pre-commit-config.yaml:8]: Was dying at the old version

File: pyproject.toml [hunk 4]
```diff
@@ -22,7 +22,7 @@ python = "^3.8"
 marshmallow = "3.13.0"
 marshmallow-dataclass = "8.5.3"
 click = "8.0.1"
-PyYAML = "5.4.1"
+PyYAML = "6.0.1"
```
Comment (lidijac) [inline pyproject.toml:25]: There was an issue with cython from last week affecting our version, options were to downgrade or upgrade

**Review by ctbur — APPROVED** [review]

**Comment by ctbur** [comment]
Thanks for the fixes!

**Pull request merged**